Kasun1Don/pump.task
# Pump 67 page heading

# Purpose
This pull request addresses the issue where the page header currently displays “Tasks Page (id)” instead of the project name. The goal is to update the header to show the actual project name.

[**Jira ticket:** PUMP-67](https://labrys-intern.atlassian.net/browse/PUMP-67?atlOrigin=eyJpIjoiZWY1NTdhMzJhZjY2NDFiZThiOGI2ZmFmNDgyMmVhZmEiLCJwIjoiaiJ9)

# Approach
- The project name is now dynamically retrieved using the `projectId` from the URL search parameters.
- The `useQuery` hook is used to fetch the project data, including the project name, from the backend via the `api.project.byId` query.
- If the project data is successfully fetched, the header will display the project’s name rather than the hardcoded "Tasks Page (id)".
- Error handling and loading states have been implemented to ensure that the correct data is displayed even while the project data is being fetched.
- Adjustments were made to ensure that the data retrieval and validation are done correctly, using `zod` for schema validation of the project data.

## Before:
![image](https://github.com/user-attachments/assets/96231a0e-6370-4186-b2e6-f19035f80a12)

## After:
![image](https://github.com/user-attachments/assets/3d7624b1-0daf-4104-8041-97f079773b72)

# Testing

1. **Ensure that the page header displays the project name instead of the project ID:**
   - Load the tasks page using a valid `projectId` in the URL parameters.
   - Check that the page header updates to show the correct project name.

2. **If the project data fails to load:**
   - Verify that a fallback message or skeleton loading state is shown while the data is being retrieved.

3. **Additional testing steps:**
   - Open the page without a valid `projectId` or an invalid `projectId`.
   - Ensure that appropriate error messages are shown (e.g., "Invalid project ID" or "Project not found").

### To test:
- Run `npm install` if any new dependencies were added.
- Ensure that the project fetching logic via the API is functional.
- Verify that the Zod validation properly handles the project data and that the correct error states are displayed when necessary.
